packages/app/components/header-dropdown.tsx

```diff
@@ -36,27 +38,51 @@ import { breakpoints } from "design-system/theme";
 type HeaderDropdownProps = {
   type: "profile" | "settings";
   withBackground?: boolean;
+  data?: UserProfile;
 };
-function HeaderDropdown({ type, withBackground = false }: HeaderDropdownProps) {
+function HeaderDropdown({
+  type,
+  withBackground = false,
+  data: user,
+}: HeaderDropdownProps) {
   const { logout } = useAuth();
   const router = useRouter();
   const { colorScheme, setColorScheme } = useColorScheme();
-  const { user } = useUser();
+  const { user: currentUser, isAuthenticated } = useUser();
   const { userAddress } = useCurrentUserAddress();
   const { width } = useWindowDimensions();
   const isWeb = Platform.OS === "web";
   const isMdWidth = width >= breakpoints["md"];
   const isDark = colorScheme === "dark";
+  const walletAddresses = useMemo(
+    () => user?.profile?.wallet_addresses_v2,
+    [user?.profile?.wallet_addresses_v2]
+  );
+  const walletAddressMatch = useMemo(
+    () => walletAddresses?.some((w) => w.address === userAddress),
+    [walletAddresses, userAddress]
+  );
+
+  // If the user is not authenticated, don't show the dropdown
+  if (!isAuthenticated) return null;
+
+  // If the user is on a profile page, and the currentUser is not the same as the passed user,
+  // don't show the dropdown menu (only show it on your own profile page)
+  if (
+    currentUser?.data.profile.username !== user?.profile.username &&
+    !walletAddressMatch
```

Comment:
curious why we also need to match the address, is the username not sufficient? 🤔

Reply:
okay, got it, new users might not have it set. Good catch 😄

Reply:
Exactly. Because you can basically skip the profile as new user and without that check you would never be able to see the gear again.